ekncontent/ekncontent/eknc-domain.c

```diff
@@ -432,22 +459,40 @@ eknc_domain_initable_init (GInitable *initable,
           g_string_append (relative_path, "../");
         }
 
-      GList *l;
-      for (l = self->subscriptions; l; l = g_list_next (l))
+      /* Import subscriptions from data directory */
+      g_autoptr(GFile) subscriptions_dir = NULL;
+      self->content_dir = eknc_get_data_dir (self->app_id);
+      subscriptions_dir = g_file_get_child (self->content_dir,
+                                            "com.endlessm.subscriptions");
+      if (g_file_query_exists (subscriptions_dir, NULL))
         {
-          const gchar *subscription_id = l->data;
-          g_autoptr(GFile) bundle_dir = g_file_get_child (subscriptions_dir,
-                                                          subscription_id);
-
-          if (!eknc_domain_process_subscription (self,
-                                                 bundle_dir,
+          if (!eknc_domain_import_subscriptions (self,
+                                                 subscriptions_dir,
                                                  json_subscriptions,
                                                  relative_path->str,
                                                  cancellable,
                                                  error))
               return FALSE;
         }
 
+      /* Import subscriptions from extensions directories */
+      g_autoptr(GList) extensions_dirs = eknc_get_extensions_dirs (self->app_id);
+      while (extensions_dirs != NULL)
+        {
+          GFile *extension_dir = (GFile *) extensions_dirs->data;
+          if (g_file_query_exists(extension_dir, cancellable))
+            {
+              if (!eknc_domain_import_subscriptions (self,
+                                                     extension_dir,
+                                                     json_subscriptions,
+                                                     relative_path->str,
+                                                     cancellable,
+                                                     error))
+                  return FALSE;
+            }
+          extensions_dirs = g_list_next(extensions_dirs);
```

Comment:
Sorry I missed this yesterday; you are leaking the list here because once extensions_dirs goes out of scope it will be pointing to NULL so there wont be anything to free.
You have to use another variable to iterate over it, this is why I prefer to use a for statement

`for (l = extensions_dirs; l; l = g_list_next(l))`